### DISCUSSION TITLE
How to load `System.Type` of a `INamedTypeSymbol`  coming from `MetadataReferenceProvider`

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/roslyn#69368] [Q&A]
Given the project dependencies:
```mermaid
graph TD
  common
  source-generator --> common
  lib --> common
  app --> lib
  app --> source-generator
```
How to reliability load types declared in `lib` from the generator code? 

This kind of works but it seems like it locks the file and prevents subsequent builds.
```cs
var assembly = Assembly.LoadFrom(compilation.GetMetadataReference(symbol.ContainingAssembly).Display)
```
Could also be my project references, I've already read through https://github.com/dotnet/roslyn/issues/52017 and https://github.com/dotnet/roslyn/discussions/47517 to make that work sometimes.

**Comment by CyrusNajmabadi** [comment]
Why do you need to load those types from your source generator?

**Reply by alrz** [reply]
The Type would be used to invoke a method (much like calling `IncrementalGenerator.Initialize` from inside the compiler).

**Reply by jasonmalinowski** [reply]
So loading assemblies like this is 100% not supported, quite simply because the library you're trying to load may have different TFM or runtime requirements than the environment you're trying to load in. If you can share your scenario more we might be able to offer alternatives.

**Reply by alrz** [reply]
> you're trying to load may have different TFM or runtime requirements than the environment you're trying to load in

Ended up just referencing it in the generator project to guarantee that. This works even if "lib" is a nuget package, except that a local generator is always needed.

Thanks.
